app/Model/CoJob.php

```diff
@@ -417,7 +463,32 @@ public function register($coId,
       // In order to present the error via the UI, we still want to insert a job record,
       // so we don't rollback.
     }
-    
+
+    // XXX Currently we support max retries only for CoreJob.Provision types
```

Comment:
Thinking about this some more, why should we special case this to Provision jobs? The max_retry parameter is on the CoJob, any type of Job could have a max retry count (even if the UI doesn't currently support setting that). This would also solve the "count" inconsistency I commented on above, we would simply track the count in co_jobs (like you already have).

Reply:
the reason i used the `ProvisioningCount` model is because i followed the CoExpirationCount paradigm.
The max_retry is the configuration. We need a way to track the actual counts and link this counts back to the model. For the Expiration Policies there is a similar field the max_count.